app/hooks/use-should-show-prompt.ts

```diff
@@ -0,0 +1,44 @@
+import { useLocalStorage } from 'usehooks-ts';
+
+type DismissedValue = 'dismissed' | number | null;
```

Comment:
I find this state to be bit unclear. I would probably make it something like:
{ state: 'dismissed' } | { state: 'dismissed-temporarily', showAfterTimestamp: number } | null